libcaf_net/caf/net/basp/message_type.hpp

```diff
@@ -0,0 +1,80 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace caf {
+namespace net {
+namespace basp {
+
+/// @addtogroup BASP
+
+/// Describes the first header field of a BASP message and determines the
+/// interpretation of the other header fields.
+enum class message_type : uint8_t {
+  /// Sends supported BASP version and node information to the server.
+  ///
+  /// ![](client_handshake.png)
+  handshake = 0x00,
```

Comment:
Is there a specific reason you are using hexadecimal here instead of decimal like you do in the other enum definitions?

Reply:
Not really. Switched to decimal.